lib/routes/account.js

```diff
@@ -333,6 +348,196 @@ module.exports = function (
+          id,
+          config.openIdVerifyUrl,
+          null, // realm
+          false, // immediate
+          false, // stateless
+          function (err, authUrl) {
+            if (err) {
+              log.error({ op: 'Account.openid.authenticate', err: err })
+              return reply.redirect(
+                config.contentServer.url + '/openid?' + qs.stringify(
+                  {
+                    err: err.message
+                  }
+                )
+              )
+            }
+            reply.redirect(authUrl)
+          },
+          openIdExtensions,
+          false // strict
+        )
+      }
+    },
+    {
+      method: 'GET',
+      path: '/account/openid/login',
+      handler: function (request, reply) {
+        if (!request.url.search) {
+          return reply(
+          '<?xml version="1.0" encoding="UTF-8"?>\n'
+          + '<xrds:XRDS xmlns:xrds="xri://$xrds" xmlns="xri://$xrd*($v*2.0)"><XRD>'
+          + '<Service xmlns="xri://$xrd*($v*2.0)">'
+          + '<Type>http://specs.openid.net/auth/2.0/return_to</Type>'
+          + '<URI>' + config.openIdVerifyUrl + '</URI>'
+          + '</Service></XRD></xrds:XRDS>'
+          ).type('application/xrds+xml')
+        }
+        log.info({ op: 'Account.openid', url: request.url })
+        openid.verifyAssertion(
+          url.format(request.url),
+          function (err, assertion) {
+            if (err || !assertion || !assertion.authenticated) {
+              log.warn({ op: 'Account.openid', err: err, assertion: assertion })
+              return reply.redirect(
+                config.contentServer.url + '/openid?err=Unknown%20Account'
+              )
+            }
+            var id = assertion.claimedIdentifier
+            var locale = request.app.acceptLanguage
+
+            if (!isOpenIdProviderAllowed(id)) {
+              log.warn({op: 'Account.openid', id: id })
+              return reply.redirect(
+                config.contentServer.url + '/openid?' + qs.stringify(
+                  {
+                    err: 'This OpenID Provider is not allowed'
+                  }
+                )
+              )
+            }
+            db.openIdRecord(id)
+              .then(
+                function (record) {
+                  return record
+                },
+                function (err) {
+                  if (err.errno !== 102) {
+                    throw err
+                  }
+                  var uid = uuid.v4('binary')
+                  var email = assertion.email || uid.toString('hex') + '@firefox.com'
+                  var authSalt = crypto.randomBytes(32)
+                  var kA = crypto.randomBytes(32)
+                  return db.createAccount(
+                    {
+                      uid: uid,
+                      createdAt: Date.now(),
+                      email: email,
+                      emailCode: crypto.randomBytes(16),
+                      emailVerified: true,
```

Comment:
> decouple sign from emailVerified :sparkles: somehow :sparkles: 

So I think this will get better when we complete the multiple-email-addresses refactor - or at least, it'll become clearer what the Right Thing is.  We can push ahead with it as-is for now.